http/middleware.go

```diff
@@ -151,3 +151,17 @@ func (b *bodyEchoer) Read(p []byte) (int, error) {
 func (b *bodyEchoer) Close() error {
 	return b.rc.Close()
 }
+
+type middleware func(http.HandlerFunc) http.HandlerFunc
+
+func handlerFuncWithMiddlewares(h http.HandlerFunc, m ...middleware) http.HandlerFunc {
```

Comment:
can you rename this to something shorter maybe? `applyMW`? this is a mouthful for something you'll be chaining.  Its quite common to see `MW` the type for middleware and how its referenced in a lot of go pkgs